contrib/openssl-cmake/CMakeLists.txt

```diff
@@ -93,6 +93,7 @@ enable_language(ASM)
 
 if(COMPILER_CLANG)
     add_definitions(-Wno-unused-command-line-argument)
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fuse-ld=lld")
```

Comment:
Setting a linker manually seems weird, tbh. This will potentially cause that different libraries of ClickHouse are linked by different linkers which is in the general case troublesome (as far as I know).

Including or not including l. 96 did not make a difference for me in local testing with static/dynamic link. I tested on x86 though, perhaps things are different on s390x.

I am not totally against explicitly specifying a linker but perhaps we can do so only for platforms where this makes a difference (s390x)?

Reply:
I observed higher Cmake file fixed the issue(which is your case), but lower version of CMake(my CMake version 3.23.2 for example) has this issue. I suggest we keep the change for lower version CMake users.

Reply:
Well, okay.